scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,128 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+    
+@pytest.mark.parametrize("dtype", DTYPES)
+@pytest.mark.filterwarnings("ignore:Casting complex")
+def test_gttrf_gttrs(dtype):
+
+    np.random.seed(42)
+    # use ?gttrf and ?gttrs to solve a random linear system for each dtype
+    n = 4
+    # set appropriate test tolerance based on the data type
+    rtol = 250*np.finfo(dtype).eps
+
+    # create the matrix in accordance with the data type
+    du = my_rand(n-1, dtype=dtype)
+    d = my_rand(n, dtype=dtype)
+    dl = my_rand(n-1, dtype=dtype)
+
+    diag_cpy = np.array([dl.copy(), d.copy(), du.copy()])
```

Comment:
This generates an object array - not simply a numerical array. Might as well just use a list.
```suggestion
    diag_cpy = [dl.copy(), d.copy(), du.copy()]
```